### ISSUE TITLE
java.io.IOException: Too many open files

### ISSUE THREAD

**Issue by sheaney** [lightbend/ssl-config#250]
There is a bug when reading PEM certificates where we could hit a `java.io.IOException: Too many open files` exception. The code is not properly closing the opened InputStream [here](https://github.com/lightbend/ssl-config/blob/master/ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/KeyStore.scala#L98-L105) and [here](https://github.com/lightbend/ssl-config/blob/master/ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/KeyStore.scala#L45-L57).

The former should look like the following maybe?
```scala
  def readCertificates(file: File): Iterable[Certificate] = {
    import scala.collection.JavaConverters._
    val cf = CertificateFactory.getInstance("X.509")
    val fis = java.nio.file.Files.newInputStream(file.toPath)
    val bis = new BufferedInputStream(fis)
    try {
      cf.generateCertificates(bis).asScala      
    } finally {
      bis.close()
    }
  }
```

**Comment by wsargent** [comment]
Looking at https://docs.oracle.com/javase/7/docs/api/java/security/cert/CertificateFactory.html#generateCertificate(java.io.InputStream) there is no guidance on whether the certificate factory closes the inputstream, although the javadocs indicate that it will consume the entire stream normally.  Sadly, the javadoc examples don't cover this explicitly.

The implementing class is `X509Factory` which does not appear to call close on the inputstream: https://github.com/AdoptOpenJDK/openjdk-jdk11u/blob/master/src/java.base/share/classes/sun/security/provider/X509Factory.java

So yes, it looks like the file instance is a bug.  Can you file a PR?

**Comment by mkurz** [comment]
Can be closed, got fixed by #251

**Comment by wsargent** [comment]
@mkurz I don't have close permissions on this repo ever since I left Lightbend :-/
